std/algorithm/iteration.d

```diff
@@ -2969,10 +2969,23 @@ iterated from the back to the front, the separator will still be consumed from
 front to back, even if it is a bidirectional range too.
  */
 auto joiner(RoR, Separator)(RoR r, Separator sep)
-if (isInputRange!RoR && isInputRange!(ElementType!RoR)
-        && isForwardRange!Separator
-        && is(ElementType!Separator : ElementType!(ElementType!RoR)))
 {
+    static assert(isInputRange!RoR, "The range of type ", RoR.stringof
+            , " is not an InputRange, aka. isInputRange!", RoR.stringof
+            , " must be true.");
+    static assert(isInputRange!(ElementType!RoR), "The ElementType!"
+            , RoR.stringof, " passed to 'joiner' does not fulfill "
+            , "isInputRange");
```

Comment:
Sorry for not being explicit, but let's make the errors consistent. What I mean is to use the same error format as the previous one. This applies for the next `static assert` also.